neo4j-to-neptune/src/test/java/com/amazonaws/services/neptune/io/Neo4jStreamWriterTest.java

```diff
@@ -0,0 +1,227 @@
+        assertNotNull("Output directory should not be null", directories.outputDirectory());
+        assertTrue("Output directory should exist", directories.outputDirectory().toFile().exists());
+        
+        // Test file path creation
+        java.nio.file.Path filePath = directories.createFilePath("test-file", "temp");
+        assertNotNull("File path should not be null", filePath);
+        assertTrue("File path should contain test-file", filePath.toString().contains("test-file"));
+        assertTrue("File path should contain temp", filePath.toString().contains("temp"));
+        assertTrue("File path should end with .csv", filePath.toString().endsWith(".csv"));
+    }
+
+    @Test
+    public void testFilePathGeneration() {
+        // Test different file path scenarios
+        java.nio.file.Path path1 = directories.createFilePath("export");
+        java.nio.file.Path path2 = directories.createFilePath("export", "1");
+        java.nio.file.Path path3 = directories.createFilePath("export", "temp");
+
+        assertNotEquals("Different paths should be generated", path1, path2);
+        assertNotEquals("Different paths should be generated", path1, path3);
+        assertNotEquals("Different paths should be generated", path2, path3);
+
+        assertTrue("All paths should end with .csv", path1.toString().endsWith(".csv"));
+        assertTrue("All paths should end with .csv", path2.toString().endsWith(".csv"));
+        assertTrue("All paths should end with .csv", path3.toString().endsWith(".csv"));
+    }
+
+    @Test
+    public void testEmptyStringValidation() {
+        // Test various empty string scenarios
+        String[] emptyStrings = {"", "   ", "\t", "\n", "\r\n"};
+        
+        for (String emptyString : emptyStrings) {
+            try {
+                new Neo4jStreamWriter(emptyString, "neo4j", "password", directories);
+                fail("Should have thrown IllegalArgumentException for empty URI: '" + emptyString + "'");
+            } catch (IllegalArgumentException e) {
+                // Expected
+                assertTrue("Error message should mention URI", 
+                    e.getMessage().toLowerCase().contains("uri"));
+            }
+        }
+    }
+
+    @Test
+    public void testValidUriFormats() {
+        // Test that various valid URI formats pass validation (will fail on connection)
+        String[] validUris = {
+            "bolt://localhost:7687",
+            "bolt+s://localhost:7687",
+            "bolt+ssc://localhost:7687",
+            "neo4j://localhost:7687",
+            "neo4j+s://localhost:7687",
+            "neo4j+ssc://localhost:7687"
+        };
+
+        for (String uri : validUris) {
+            try {
+                new Neo4jStreamWriter(uri, "neo4j", "password", directories);
+                fail("Should have thrown RuntimeException due to connection failure for URI: " + uri);
+            } catch (RuntimeException e) {
```

Comment:
Would be good to assert that e is not an IllegalArgumentException thrown from the validation as IllegalArgumentException is a subclass of RuntimeException